merlin/models/tf/inputs/embedding.py

```diff
@@ -318,135 +327,132 @@ def compute_call_output_shape(self, input_shapes):
+    if activity_regularizer:
+        kwargs["activity_regularizer"] = activity_regularizer
+    if sequence_combiner:
+        kwargs["sequence_combiner"] = sequence_combiner
 
-    cols = schema.select_by_tag(Tags.CATEGORICAL)
     tables = {}
 
-    embedding_dims = embedding_dims or {}
-    if infer_embedding_sizes:
-        inferred_embedding_dims = schema_utils.get_embedding_sizes_from_schema(
-            schema,
-            infer_embedding_sizes_multiplier,
-            infer_embeddings_ensure_dim_multiple_of_8,
-        )
-        # Adding inferred embedding dims only for features where the embedding sizes
-        # were not pre-defined
-        inferred_embedding_dims = {
-            k: v for k, v in inferred_embedding_dims.items() if k not in embedding_dims
-        }
-        embedding_dims = {**embedding_dims, **inferred_embedding_dims}
-
-    trainable = trainable or {}
-    for col in cols:
-        combiner = None
-        if Tags.SEQUENCE in col.tags or Tags.LIST in col.tags or col.is_list:
-            if isinstance(sequence_combiner, dict):
-                combiner = sequence_combiner[col.name]
-            else:
-                combiner = sequence_combiner
+    for col in schema:
+        table_kwargs = _forward_kwargs_to_table(col, table_cls, kwargs)
+        tables[col.name] = table_cls(_get_dim(col, dim, infer_dim_fn), col, **table_kwargs)
+
+    return ParallelBlock(
+        tables, pre=pre, post=post, aggregation=aggregation, name=block_name, schema=schema
+    )
 
-        embedding_size = embedding_dims.get(col.name, embedding_dim_default)
 
-        if embeddings_initializers:
-            if isinstance(embeddings_initializers, dict):
-                emb_initializer = embeddings_initializers.get(col.name, "uniform")
+def _forward_kwargs_to_table(col, table_cls, kwargs):
+    arg_spec = inspect.getfullargspec(table_cls.__init__)
+    table_kwargs = dict(zip(arg_spec.args[-len(arg_spec.defaults) :], arg_spec.defaults))
+
+    for key, val in kwargs.items():
+        if key in table_kwargs:
+            if isinstance(val, dict):
+                if col.name in val:
+                    table_kwargs[key] = val[col.name]
             else:
-                emb_initializer = embeddings_initializers
-            kwargs["embeddings_initializer"] = emb_initializer
-
-        tables[col.name] = EmbeddingTable(
-            embedding_size,
-            col,
-            combiner=combiner,
-            trainable=trainable.get(col.name, True),
-            **kwargs,
-        )
+                table_kwargs[key] = val
 
-    return ParallelBlock(
-        tables, pre=pre, post=post, aggregation=aggregation, name=block_name, schema=cols
-    )
+    return table_kwargs
+
+
+def _get_dim(col, embedding_dims, infer_dim_fn):
+    dim = None
+    if isinstance(embedding_dims, dict):
+        dim = embedding_dims.get(col.name)
+    elif isinstance(embedding_dims, int):
+        dim = embedding_dims
+
+    if not dim:
```

Comment:
@marcromeyn Before we were able to define embedding dims to specific features and infer embeddding dims for all other features. But now it seems you can either set dims or infer them, but not both. 
That was useful for example for the retrieval experiments, so that we can hypertune the user and item dims independently (so that capacity of TwoTower and MF is comparable).

Reply:
@gabrielspmoreira it should only reach this line (to infer the `dim` parameter) if it's not already specified. The difference is that all the args can now be specified as a single value (`T` - int in this case) or `Dict[str, T]`. 

Example below of specifying a dim for one column (`item_id`), leaving the rest to be inferred (`item_feature`)

```python
import pyarrow as pa
import numpy as np
import merlin.models.tf as mm
from merlin.schema import Schema, ColumnSchema, Tags

schema = Schema(
    [
        ColumnSchema(
            "item_id",
            properties={"domain": {"min": 0, "max": 10}},
            dtype=np.int32,
        ),
        ColumnSchema(
            "item_feature",
            properties={"domain": {"min": 0, "max": 10}},
            dtype=np.int32,
        ),
    ]
)


embeddings = mm.Embeddings(schema, dim={"item_id": 4})

embeddings["item_id"].table.output_dim       # => 4 (specified)
embeddings["item_feature"].table.output_dim  # => 8 (inferred)
```